### DISCUSSION TITLE
XPath + HTML title

### DISCUSSION THREAD

**Discussion by pthall** [FreshRSS/FreshRSS#6923] [Q&A] [answered]
### Describe the bug

The XPath for items and item title has been causing me immense confusion.

#### This items XPath works:
```
//div[@class="we-truncate we-truncate--multi-line  targeted-link__target"]
```
#### with this item title XPath:
```
descendant::div
```
#### and this item content XPath:
```
.
```

#### items XPath:
```
//div[@class="we-clamp we-clamp--visual"]
```
#### with this item title XPath:
```
.
```
#### and this item content XPath:
```
.
```

### To Reproduce

The XPaths and URL described the steps to reproduce.

### Expected behavior

It should be possible to use `.` or `self` for title as valid XPath and not be required to specify `descendant::`.

### FreshRSS version

1.24.3

### Environment information

- Database version: Postgres 15.8
- PHP version: PHP 8.3
- Installation type: Google Cloud VM
-Web server type: Apache
- Device: Windows 11
- OS: Debian 12
- Browser: Firefox 131

**Comment by Alkarex** [comment, marked as answer]
Hello,
Try to curl `https://apps.apple.com/us/developer/insulet-corporation/id1101349399`: there is no element with a class equal to `"we-clamp we-clamp--visual"`
Watch out that if you look at the DOM from your Web browser developer tools, as it may have been changed dynamically by JavaScript. You need to look at the untransformed source code.

**Reply by pthall** [reply]
Thank you! Scraping is a fantastic feature of FreshRSS.
Do you have suggestions on tooling when scraping XPath + HTML? Browser JS console is nice for interactivity, unfortunately it does not reflect the non-JS rendered environment of FreshRSS.

**Reply by Alkarex** [reply]
You can keep using the tools of your Web browser but after disabling JavaScript.
Alternatively, open the downloaded source code in a code editor, so you can auto-format, search, collapse, etc.
